modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceDeployerApi.java

```diff
@@ -155,9 +156,10 @@ public List<DeployedService> listDeployedServicesDetails(
             @RequestParam(name = "serviceVersion", required = false) String serviceVersion,
             @Parameter(name = "serviceState", description = "deployment state of the service")
             @RequestParam(name = "serviceState", required = false)
-                    ServiceDeploymentState serviceState) {
+            ServiceDeploymentState serviceState) {
         // return type is DeployedService but actually returns one of the child types
         // VendorHostedDeployedServiceDetails or DeployedServiceDetails
+        cspPluginValidator.validatePluginEnabled(csp);
```

Comment:
Discussing how to get this validation automatically done for each method using AOP.

Reply:
Removed all of this validation for request body  into the AOP and added the validation method into the `CspEnumConverter` for `csp`  in the URI.